getExclusions.m

```diff
@@ -11,49 +10,52 @@
                 tcs.Move.(types{1}));
 nUnits = tmp(:, 1);                             % how many units per area                    
 
-% preallocate
-for ia = 1:nAreas
-   OK.isVisual{ia} = zeros(nTypes, nUnits(ia));   
-   OK.anyVisual{ia} = zeros(1, nUnits(ia));   
-   OK.DSI{ia} = zeros(nTypes, nUnits(ia));
-end
-
-
 % Must respond more to move than blank throughout experiment
 %   in at least one of the 12 directions
 
 % parameters 
 pThresh = 0.05;
-nBins = 4;
-
+trialStep = 10; %test 10 trials at a time for visual responsiveness (move > blank)
 
 for ia = 1:nAreas   
 for it = 1:nTypes
-    overThresh = zeros(nUnits(ia), nDirs);
+    p = nan(nUnits(ia), nBins, nDirs);
 for id = 1:nDirs
     tmpMV = tcs.Move.(types{it}){ia,id};
     tmpBK = tcs.Blank.(types{it}){ia,id};
     
     nTrials = size(tmpBK, 2);
-    trialStep = nTrials/nBins;
+    nBins = ceil(nTrials/trialStep);
     binStarts = 1:trialStep:nTrials;
+    
+    %preallocate
+    if it == 1 && id == 1
+        OK.byTime{ia} = zeros(nTypes, nUnits(ia), nTrials);
+        OK.isVisual{ia} = zeros(nTypes, nUnits(ia));   
+        OK.anyVisual{ia} = zeros(1, nUnits(ia));   
+        OK.DSI{ia} = zeros(nTypes, nUnits(ia));
+    end
+    
     for iu = 1:nUnits(ia)
-        p = zeros(1, nBins);
         for ib = 1:nBins
             trialRange = binStarts(ib):binStarts(ib)+trialStep-1;
-            p(ib) = kruskalwallis([tmpBK(iu,trialRange)' ...
-                                   tmpMV(iu,trialRange)'], [], 'off');
+            trialRange = trialRange(trialRange <= length(tmpBK)); %last bin may have fewer trials
+            p(iu, ib, id) = ranksum(tmpBK(iu,trialRange), ...
+                                    tmpMV(iu,trialRange), 'tail', 'left');
```

Comment:
Why only (...,'tail','left')? 
Thinking generically, we might just care about significant changes.

Reply:
I don't think it's likely that the move period will reduce firing rate, and wanted more statistical power.

Reply:
OK, given that this is an experiment-specific script, that's fine.